--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -88,6 +88,15 @@ define Device/friendlyarm_nanopi-r2s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r2s
 
+define Device/friendlyarm_nanopi-r2s-plus
+  $(call Device/friendlyarm_nanopi-r2s)
+  DEVICE_MODEL := NanoPi R2S Plus
+  DEVICE_VARIANT := eMMC
+  DEVICE_DTS := rockchip/rk3328-nanopi-r2s-plus
+  SUPPORTED_DEVICES += friendlyarm,nanopi-r2s-plus
+endef
+TARGET_DEVICES += friendlyarm_nanopi-r2s-plus
+
 define Device/friendlyarm_nanopi-r4s
